### ISSUE TITLE
verify-ucm-config.sh: duplicate of alsa-tests/ucm-validator

### ISSUE THREAD

**Issue by plbossart** [thesofproject/sof-test#240]
In alsa-test, there is an ucm-validator already. Do we need to keep this script or should we switch to the standard mainline test?

**Comment by xiulipan** [comment]
@libinyang Please help to answer this question.

**Comment by libinyang** [comment]
My understand ucm-validator only checks ucm generally. It won't check the contents is correct or not on a special platform. And verify-ucm-config. sh will check whether the ucm works on the special platform, such as whether the amixer setting can be executed on the platform or not, as well as whether the UCM grammar is correct or not.

**Comment by plbossart** [comment]
@libinyang ucm-validator will scan all branches of a UCM file and check if all tests are covered and the syntax is correct.

That seems to do most of what you described, and actually you will only test the branches that are taken on a given platform.

**Comment by libinyang** [comment]
Sorry for the delay reply. I studied the ucm-validator and tried to understand it well. 

verify-ucm-config.sh will not only checks the syntax, but also will check such typo error "cset "name='wrong kctl name' value". But it seems ucm-validator won't. This can help us to identify such silly issue quickly. If we don't care such issues, I think we can remove verify-ucm-config.sh. 

And I'm trying to use the cmd 'ucm.py' to check the ucm configuration with 'one' or 'all' parameter, e.g. "./ucm.py one  /usr/share/alsa/ucm2/sof-soundwire/sof-soundwire.conf" or "./ucm.py all  /usr/share/alsa/ucm2/". But both failed. @plbossart , do you know what's wrong with my command?

**Comment by plbossart** [comment]
> Sorry for the delay reply. I studied the ucm-validator and tried to understand it well.
> 
> verify-ucm-config.sh will not only checks the syntax, but also will check such typo error "cset "name='wrong kctl name' value". But it seems ucm-validator won't. This can help us to identify such silly issue quickly. If we don't care such issues, I think we can remove verify-ucm-config.sh.

I don't get your point. What does 'wrong kctrl name'? A control that does not exist? A control that is used instead of another one?

> And I'm trying to use the cmd 'ucm.py' to check the ucm configuration with 'one' or 'all' parameter, e.g. "./ucm.py one /usr/share/alsa/ucm2/sof-soundwire/sof-soundwire.conf" or "./ucm.py all /usr/share/alsa/ucm2/". But both failed. @plbossart , do you know what's wrong with my command?

	$(DBGCMD) ./ucm.py --level $(V) configs $(ALSA_UCM_DIR) configs $(CONFIGS_FILTER)

You seem to be missing the configs?

**Comment by libinyang** [comment]
> > Sorry for the delay reply. I studied the ucm-validator and tried to understand it well.
> > verify-ucm-config.sh will not only checks the syntax, but also will check such typo error "cset "name='wrong kctl name' value". But it seems ucm-validator won't. This can help us to identify such silly issue quickly. If we don't care such issues, I think we can remove verify-ucm-config.sh.
> 
> I don't get your point. What does 'wrong kctrl name' mean? A control that does not exist? A control that is used instead of another one?

Yes, I mean a control doesn't exist. Sometimes, many boards will share a same config. If we create the config carelessly, we may use a kctl which exists on one board but not exists on another board.

> ```
> $(DBGCMD) ./ucm.py --level $(V) configs $(ALSA_UCM_DIR) configs $(CONFIGS_FILTER)
> ```
> 
> You seem to be missing the configs?

Do you mean the files in the folder "ucm-validator/configs/sof-soundwire/"? I have such configs in the folder.
I find if I run your suggested command or run `make configs` (the two command are doing the same thing) in the  "ucm-validator" folder, it works.
But if I run `make verify`, it fails.

**Comment by libinyang** [comment]
@plbossart I suggest keeping this case. All other test cases including ucm-validator won't detect this 'cset' issue. Even with manually paplay, we can't find this issue too. Without this case, we have to scrutinizing the pulseaudio log or manually using sound-setting to detect such issues. What do you think? If it is OK to you, may I close this issue?

**Comment by plbossart** [comment]
@libinyang even if we kept the verify-ucm-config, it doesn't test the boot sequence, does it?

**Comment by libinyang** [comment]
> @libinyang even if we kept the verify-ucm-config, it doesn't test the boot sequence, does it?

@plbossart No, it doesn't. Thanks for reminder. I will add this case.
